### DISCUSSION TITLE
Perplexica 1.9.0

### DISCUSSION THREAD

**Discussion by ItzCrazyKns** [ItzCrazyKns/Vane#406] [Announcements]
Perplexica 1.9.0 is finally out! It brings the awaited discover page that allows you to chat with the trending topics. What do you think about it?

**Comment by dvejsada** [comment]
Do you plan to update docs and docker-compose.yml to point to pre-build docker images?

**Reply by ItzCrazyKns** [reply]
The docs are updated and the prebuilt images are used by default in the compose file.

**Reply by dvejsada** [reply]
Thanks, but unfortunately, the IP addresses are still hard coded in the prebuilt images, so I am unable to change them when using these images (and they do not work). I tried to add similar env variables to docker compose, but it will not overwrite the initially provided ones.

**Reply by ItzCrazyKns** [reply]
Please follow the networking guide: https://github.com/ItzCrazyKns/Perplexica/blob/master/docs/installation/NETWORKING.md

**Reply by ngdbao** [reply]
I think what he means is a pre-built docker, which can inject IP address by mount config, not re-build at all

**Reply by dvejsada** [reply]
> Please follow the networking guide: https://github.com/ItzCrazyKns/Perplexica/blob/master/docs/installation/NETWORKING.md

I understand this, but this requires to rebuild the images, you cannot use the prebuilt ones. Why not to remove these args when building images and set them as env variables later when running the container? Maybe I am missing something,sorry...

**Reply by ItzCrazyKns** [reply]
While building the images, the Next JS app is built. Next JS bundles the environment variables (the public ones) into the javascript code it generates so there's no way I could change it rather than the user building the images themselves.

**Reply by dvejsada** [reply]
> While building the images, the Next JS app is built. Next JS bundles the environment variables (the public ones) into the javascript code it generates so there's no way I could change it rather than the user building the images themselves.

Thanks for the explanation. I have quickly glanced over the Next.js docs which mention that also runtime env. variables are supported (i.e. not bundled at build time), specifically for the case of having one Docker image for different environments. But I do not know if this is possible to implement for this specific case.

**Comment by lludlow** [comment]
I really like the Dicover feature. I have been playing around adding different sources and search terms. Are there plans to add a UI element to modify these?

**Reply by ItzCrazyKns** [reply]
Yes, I have it planned

**Reply by onelouder** [reply]
In the mean time you can edit  /src/app/api/discover/route.ts by 
changing the search sites and topics:
```
const articleWebsites = [
  'yahoo.com',
  'www.exchangewire.com',
  'businessinsider.com',
  /* 'wired.com',
  'mashable.com',
  'theverge.com',
  'gizmodo.com',
  'cnet.com',
  'venturebeat.com', */
];
```

**Comment by gerroon** [comment]
How is the discovery page populated? Is there a way to edit the sources?

**Reply by onelouder** [reply]
Discover web news source URLs and topics are specified in src/app/api/discover/route.ts.    You could modify the urls and the topic list, but the comment mentions this is a future plan.
